src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -810,6 +819,40 @@ func (r *StatefulSetResource) getPorts() []corev1.ContainerPort {
 	return ports
 }
 
+func (r *StatefulSetResource) getStartupProbe() *corev1.Probe {
+	adminApi := r.pandaCluster.AdminAPIInternal()
+	if adminApi == nil {
+		r.logger.Info("BUG! an internal listener for admin API is required")
+		return nil
+	}
+	// TODO: add support for installations with TLS on internal admin API
+	if adminApi.TLS.Enabled || adminApi.TLS.RequireClientAuth {
+		r.logger.Info("WARNING! with TLS enabled for the internal adminAPI, the operator won't add a Startup probe to Redpanda nodes which might be problematic during upgrades or when adding nodes to the cluster")
+		return nil
+	}
+
+	baseProbe := corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "v1/status/started",
+				Port:   intstr.FromInt(adminApi.Port),
+				Scheme: "HTTP",
+			},
+		},
+		// TODO: make max retries configurable
+		//
+		// For clusters with large amounts of data, the startup of a new node can take 10s of minutes.
+		// As a measure of precaution, we're setting the min time for the probe to fail to ~1h.
+		InitialDelaySeconds: 1,
+		TimeoutSeconds:      1,
+		PeriodSeconds:       1,
+		SuccessThreshold:    1, // Once the node is in "started" state, it won't go back to "booting" state
+		FailureThreshold:    3000,
```

Comment:
What is the k8s behavior if the failure threshold of 3000s (50m) is reached, would the containers restart?

Which variables or parameters determine the time it takes for Redpanda's endpoint (`v1/status/started`) to return success? I'd think the amount of data on the machine is one of them.

To confirm, are non-started redpanda instances available over the network?

Reply:
> What is the k8s behavior if the failure threshold of 3000s (50m) is reached, would the containers restart?

Yeah, all failing probes force restarts - that's their primary goal.

> Which variables or parameters determine the time it takes for Redpanda's endpoint (`v1/status/started`) to return success? I'd think the amount of data on the machine is one of them.

A single request is sent to each Redpanda broker that has leaders for partitions the queried node has replicas of (assume all of them). Each request is sent in parallel. Each request loops over all partitions on the target, but the results are pretty much from cache.

> To confirm, are non-started redpanda instances available over the network?

The admin endpoint listens pretty early on in the lifecycle of Redpanda.

Reply:
Thanks, Ben. My overall question is whether there is any Redpanda state from which it is currently possible to transition to a working state, but the probe would prevent us from doing so.

Btw. what is the advantage of a non-infinite failure threshold, that the number of pod restarts increases?